Dockerfile.Windows

```diff
@@ -1,9 +1,25 @@
-FROM mcr.microsoft.com/windows/servercore:1809 as core
+# Other common ones: 1803-amd64, 1903-amd64
+# full list: curl -L https://mcr.microsoft.com/v2/windows/nanoserver/tags/list
+
+ARG golangTag=1.14.0-windowsservercore-1809
+ARG nanoserverTag="mcr.microsoft.com/windows/nanoserver:1809"
+ARG servercoreTag="mcr.microsoft.com/windows/servercore:1809"
+ARG goflags_vendor=""
+ARG buildVersion=""
+
+FROM golang:${golangTag} as builder
+ENV SRC_PATH="src/github.com/kubernetes-csi/node-driver-registrar"
+COPY . ${SRC_PATH}
+RUN cd $ENV:SRC_PATH ; go build ${goflags_vendor} -a -ldflags '-X main.version=${buildVersion} -extldflags "-static"' -o ./bin/csi-node-driver-registrar.exe ./cmd/csi-node-driver-registrar ; ls ./bin ;
```

Comment:
Currently, it's possible to build the binaries separately and then construct the container image with it. After this change, is there a way we can still do it ?

Reply:
Yes this would eliminate that in favor of of a single `docker build` command. The 1 thing I am not crazy about is piping in the `buildVersion`. I would like to find a better way to fetch that in the dockerfile but my windows scripting knowledge isnt all that great. I figured this was a good first step and as we do more of these I will find a better way to get this information at build time and I can refactor then.

Reply:
This is different from how the linux Dockerfile works. I would like to keep the behavior consistent. Can we somehow make this optional, or have another Dockerfile?

Reply:
You have no managed way to build the exe other than the makefile which wont work on windows. How do you expect your users to build this image today?

Reply:
You will have to build this image on windows. Doing it this way provides a singular way to do that which is managed by the repo. If you require another system to build the binary before it can build the image then you need to provide a way to build that binary that is consistent with how you build the linux binary but able to do so on said operating system.

Reply:
> ... the makefile which wont work on windows

Why? `1.14.0-windowsservercore-1809` seems to have a POSIX shell and the corresponding tools (like `ls`). Is there no POSIX-compatible make?

The current Makefile expects also "git" for determining the version, but perhaps even that can be made available?

Reply:
alright so i should have been a bit more clear. its the fact taht the make targets call out to bash scripts that make this an issue. 

```shell
PS C:\gopath\src\github.com\kubernetes-csi\node-driver-registrar> make
./release-tools/verify-go-version.sh "go"
/usr/bin/env: 'bash': No such file or directory
make: *** [release-tools/build.make:194: check-go-version-go] Error 127
```

Reply:
@pohly can we decouple the "make" rule from verifying go version?

I want to share as much common config and rules as we can so that they don't get out of sync with each other.

Reply:
> bash scripts that make this an issue

Is plain POSIX acceptable? The `#!/usr/bin/env bash` stanca was copied from some other scripts. `verify-go-version.sh` should also work with just `#!/bin/sh`.

> can we decouple the "make" rule from verifying go version?

We could add an env variable for suppressing it, but in this particular case the Go version check is useful because it will trigger when `golangTag` gets out of sync. I'd prefer to make the check work.

Removing it from `make build` unconditionally would defeat the purpose of the check, because then developers also no longer get the warning.

Reply:
really it should be os dependent and just call the right script. powershell vs bash in this case

Reply:
The problem then is that we have to maintain two version of this script and potentially others. I don't know powershell, so I can't create or maintain anything written in it.

POSIX has been around for a while. If there is a way to support it also for Windows builds, then I vote for enabling that and writing cleaner POSIX-compatible scripts that then also run on Windows.

Reply:
@PatrickLang or @marosset would probably be able to shed a bit more light on this as I am definitely not a windows expert here.